### PR TITLE
feat: add monitorv2 datasource

### PR DESCRIPTION
**Pull request by obs-gh-owengoebel**

### CODE REVIEW

File: client/api.go [hunk 1]
```diff
@@ -525,6 +525,10 @@ func (c *Client) GetMonitorV2(ctx context.Context, id string) (*meta.MonitorV2,
 	return c.Meta.GetMonitorV2(ctx, id)
 }
 
+func (c *Client) LookupMonitorV2(ctx context.Context, workspaceId *string, folderId *string, nameExact *string, nameSubstring *string) (*meta.MonitorV2, error) {
```
Comment (vikramraman) [inline client/api.go:528]: Even if the actual api has all these fields, we probably only want to support workspaceId and nameExact in here.

File: observe/data_source_monitor_v2.go [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+func dataSourceMonitorV2() *schema.Resource {
+	return &schema.Resource{
+		Description: descriptions.Get("monitorv2", "description"),
+		ReadContext: dataSourceMonitorV2Read,
+		Schema: map[string]*schema.Schema{
+			// used to lookup the monitor
+			// monitor can be looked up either by providing an ID
+			// or by providing search params that can uniquely ID the monitor.
+			"get_id": { // ObjectId!
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:23]: We want to be consistent with the naming across data sources, we typically have: `id`, `workspace`, `name` as the fields for looking up stuff, and should go with the same here.

File: observe/data_source_monitor_v2.go [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateDiagFunc: validateID(),
+				Description:      descriptions.Get("common", "schema", "id"),
+			},
+			"search_workspace_id": { // ObjectId!
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:29]: `workspace`

File: observe/data_source_monitor_v2.go [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.Get("monitorv2", "schema", "workspace_id"),
+			},
+			"search_name_exact": {
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:34]: `name`

File: observe/data_source_monitor_v2.go [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// fields of MonitorV2Input excluding the components of MonitorV2Definition
+			"workspace_id": { // ObjectId!
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:39]: Can be removed, it is redundant with the `workspace` field.

File: observe/data_source_monitor_v2.go [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.Get("monitorv2", "schema", "comment"),
+			},
+			"rule_kind": { // MonitorV2RuleKind!
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.Get("monitorv2", "schema", "rule_kind"),
+			},
+			"name": { // String!
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:54]: same, redundant given `name` above

File: observe/data_source_monitor_v2.go [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+										Computed:    true,
+										Description: descriptions.Get("monitorv2", "schema", "scheduling", "transform", "freshness_goal"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"id": { // ObjectId!
```
Comment (vikramraman) [inline observe/data_source_monitor_v2.go:260]: We want to include an `oid` instead as a computed field.

**Review by vikramraman — CHANGES_REQUESTED** [review]

File: docs/data-sources/monitor_v2.md [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+template and destinations to configure the receiver.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `data_stabilization_delay` (String) expresses the minimum time that should elapse before data is considered "good enough" to evaluate. Choosing a delay really depends on the expectations of latency of data and whether data is expected to arrive later than other data and thus would change previously evaluated results.
```
Comment (vikramraman) [inline docs/data-sources/monitor_v2.md:29]: This looks odd, it should be computed and not be part of the optional input for this data source

File: docs/data-sources/monitor_v2.md [hunk 9]
```diff
@@ -0,0 +1,411 @@
...
+- `id` (String) Resource ID for this object.
+- `lookback_time` (String) optionally describes a duration that must be satisifed by this monitor. It applies to all rules, but is only applicable to rule kinds that utilize it.
```
Comment (vikramraman) [inline docs/data-sources/monitor_v2.md:31]: Similar comment here, this should not be needed

**Review by vikramraman — CHANGES_REQUESTED** [review]

File: docs/data-sources/monitor_v2.md (hunk 8 above)
Comment (obs-gh-owengoebel) [reply docs/data-sources/monitor_v2.md]: makes sense. I think I forgot to delete "Optional: true" when adding "Computed: true". Fixed now

**Review by vikramraman — APPROVED** [review]

**Pull request merged**